app/controllers/canvas_mailing_lists_controller.rb

```
@@ -23,9 +23,11 @@ def show
   end
 
   # POST /api/academics/canvas/mailing_lists/:canvas_course_id/create
+  # Unlike other CRUD actions, list creation must specify which type of list is being created.
 
   def create
-    list = MailingLists::SiteMailingList.create canvas_site_id: canvas_course_id.to_s, list_name: params['listName']
+    list_class = "MailingLists::#{params['listType'] || 'CalmailList'}".constantize
```

@paulkerschen any concern about `params['listType']` containing an invalid name (or the name of a class you don't want to expose)?  Or will `constantize` throw a `NameError` if the class is not defined?